store/displayPreferences.ts

```diff
@@ -0,0 +1,197 @@
+  displayPreferences: undefined
+});
+
+interface SingleCustomPrefMutationPayload {
+  key: string;
+  value: string;
+}
+
+export const getters: GetterTree<
+  DisplayPreferencesState,
+  DisplayPreferencesState
+> = {
+  /**
+   * Returns the CustomPrefs object
+   *
+   * @param {DisplayPreferencesState} state Current state
+   * @returns {object | null | undefined} CustomPrefs object
+   */
+  getCustomPrefs: (state: DisplayPreferencesState) =>
+    state.displayPreferences?.CustomPrefs,
+
+  /**
+   * Gets the current dark mode setting as a boolean (default is enabled)
+   *
+   * @param {DisplayPreferencesState} state Current state
+   * @returns {boolean} Current dark mode
+   */
+  getDarkMode: (state: DisplayPreferencesState) =>
+    !(
+      state.displayPreferences != null &&
+      state.displayPreferences.CustomPrefs != null &&
+      state.displayPreferences.CustomPrefs.darkMode === 'False'
+    )
+};
+
+export const mutations: MutationTree<DisplayPreferencesState> = {
+  /**
+   * Sets the internal state with the server answer
+   *
+   * @param {DisplayPreferencesState} state Current state
+   * @param {any} param1 Payload
+   * @param {DisplayPreferencesDto} param1.displayPreferences Display preferences returned by the server
+   */
+  INIT_STATE(
+    state: DisplayPreferencesState,
+    { displayPreferences }: { displayPreferences: DisplayPreferencesDto }
+  ) {
+    state.displayPreferences = displayPreferences;
+  },
+
+  /**
+   * Edits a custom pref key
+   *
+   * @param {DisplayPreferencesState} state Current state
+   * @param {any} param1 Payload
+   * @param {SingleCustomPrefMutationPayload} param1.pref Key and value for new custom pref
+   */
+  EDIT_CUSTOM_PREF(
+    state: DisplayPreferencesState,
+    { pref }: { pref: SingleCustomPrefMutationPayload }
+  ) {
+    if (state.displayPreferences == null) return;
+    if (state.displayPreferences.CustomPrefs == null)
+      state.displayPreferences.CustomPrefs = {};
+    state.displayPreferences.CustomPrefs[pref.key] = pref.value;
+  },
+
+  /**
+   * Drops a custom pref key
+   *
+   * @param {DisplayPreferencesState} state Current state
+   * @param {any} param1 Payload
+   * @param {string} param1.key Key to drop
+   */
+  DROP_CUSTOM_PREF(state: DisplayPreferencesState, { key }: { key: string }) {
+    if (
+      state.displayPreferences == null ||
+      state.displayPreferences.CustomPrefs == null
+    )
+      return;
+    delete state.displayPreferences.CustomPrefs[key];
+  }
+};
+
+export const actions: ActionTree<
+  DisplayPreferencesState,
+  DisplayPreferencesState
+> = {
+  /**
+   * Fetches display preferences, stores them and updates Vuetify dark mode
+   *
+   * @param {any} param0 Vuex
+   * @param {any} param0.commit Vuex commit
+   * @param {any} param0.dispatch Vuex dispatch
+   */
+  async initState({ commit, dispatch }) {
+    const response = await this.$api.displayPreferences.getDisplayPreferences({
```

Comment:
You can try to catch this `api` call cause it can fail and we won't understand it until we open the console.

https://github.com/jellyfin/jellyfin-vue/blob/master/store/user.ts#L43 `loginRequest` is an example of how you can proceed further

Reply:
Yeah that's fair enough. What should I do in case of failure? Single snackbar? For now it's not a problem as only the dark mode is used, I could say "Can't load dark mode preferences, applying the default one". But what will it be in the future if more settings are used?

Reply:
Snackbar is probably enough, for now.

Something like "Failed to load display preferences". We should have a copy of the object this call returns as the default state, so we have sensible defaults to play with in case of a failure.

Reply:
I like that, I'll do that

Reply:
Done, there's a default value for dark mode and a check for the status of the answer

Reply:
I would encourage you to set a 'throw Error()' inside the 'getDisplayPreferences' method and observe what the action code will do.

Reply:
I'd prefer avoiding meddling with the API files as they are not in the codebase anymore. But I can wrap my call and code check in the try

Reply:
How do you like what I've done?

Reply:
Much better! Now we will catch an error response from the server. Can we proceed even further with following the pattern that the other stores used for the API calls? I mean, having a `success` action and `failure` action for the calls. `loginRequest` is still an example of what I am referring to.

Reply:
I don't know about that, they do not act the same. One is for getting data, the other one is just to push. I've put the error handling in its own action, is it enough? :)

Reply:
What do you mean by ‘they do mot act the same’? One makes the request, one handles the success of this request and one handles the error of it. The main idea is that at some point you would want to know when the preferences are loaded. You can do that by subscribing to the store and listen for the success action instead of ‘try catch’ the ‘initState’

Reply:
Ok even though I do feel it's kinda bloaty, I agree with the ease of use to subscribe to the methods. So I've done it with the two methods which query the server. Is it how you were thinking?